### PR TITLE
[New Rule] File System Debugger ‘debugfs’ Launched Inside a Privileged Container

### PR DESCRIPTION
**Pull request by imays11**
## Issues

https://github.com/elastic/detection-rules/issues/3228

## Summary

This rule detects the use of the built-in Linux DebugFS utility from inside a privileged container. DebugFS is a special file system debugging utility which supports reading and writing directly from a hard drive device. When launched inside a privileged container, a container deployed with all the capabilities of the host machine, an attacker can access sensitive host level files which could be used for further privilege escalation and container escapes to the host machine.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+a privileged container, a container deployed with all the capabilities of the host machine, an attacker can access
+sensitive host level files which could be used for further privilege escalation and container escapes to the host
+machine.
+"""
+from = "now-6m"
+index = ["logs-cloud_defend*"]
+interval = "5m"
+language = "eql"
+license = "Elastic License v2"
+name = "File System Debugger �debugfs� Launched Inside a Privileged Container"
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:23]: Not sure whether my client just can't render this name or there is something odd in the rule name..

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.module == "cloud_defend" and     
+  event.type== "start" and process.name== "debugfs" and 
+  container.security_context.privileged == true
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:44]: Maybe you can make this one more specific. I created a similar rule for privilege escalation a while ago https://github.com/elastic/detection-rules/blob/main/rules/linux/privilege_escalation_sda_disk_mount_non_root.toml and noticed that the `-R` is often used for benign purposes, and that you require to specifiy a disk device before being able to access it. Might be good constraints!

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml (hunk 1 above)
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:23]:
```suggestion
name = "File System Debugger Launched Inside a Privileged Container"
```

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml (hunk 2 above)
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:44]:
```suggestion
process where event.module == "cloud_defend" and     
  event.type== "start" and process.name== "debugfs" and 
  process.args : "/dev/sd*" and not process.args == "-R" and
  container.security_context.privileged == true
```

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 3]
```diff
@@ -0,0 +1,60 @@
+[metadata]
+creation_date = "2023/10/26"
+integration = ["cloud_defend"]
+maturity = "production"
+min_stack_comments = "New Integration: Cloud Defend"
+min_stack_version = "8.8.0"
+updated_date = "2023/10/26"
```
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:7]:
```suggestion
updated_date = "2023/12/18"
```

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+index = ["logs-cloud_defend*"]
+interval = "5m"
+language = "eql"
+license = "Elastic License v2"
+name = "File System Debugger Launched Inside a Privileged Container"
+references = [
+    "https://cyberark.wistia.com/medias/ygbzkzx93q?wvideo=ygbzkzx93q",
+    "https://book.hacktricks.xyz/linux-hardening/privilege-escalation/docker-security/docker-breakout-privilege-escalation#privileged",
+]
+risk_score = 73
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:28]: Might be good to start with a medium risk score, and move up in promotion when we are sure that there are limited FPs!

**Review by Aegrah — APPROVED** [review]

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    "OS: Linux",
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+process where event.module == "cloud_defend" and     
+  event.type== "start" and process.name== "debugfs" and
```
Comment (Aegrah) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:43]:
```suggestion
  event.type == "start" and process.name == "debugfs" and 
```

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml (hunk 4 above)
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:28]:
```suggestion
risk_score = 47
```

File: rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+language = "eql"
+license = "Elastic License v2"
+name = "File System Debugger Launched Inside a Privileged Container"
+references = [
+    "https://cyberark.wistia.com/medias/ygbzkzx93q?wvideo=ygbzkzx93q",
+    "https://book.hacktricks.xyz/linux-hardening/privilege-escalation/docker-security/docker-breakout-privilege-escalation#privileged",
+]
+risk_score = 73
+rule_id = "97697a52-4a76-4f0a-aa4f-25c178aae6eb"
+severity = "high"
```
Comment (imays11) [inline rules/integrations/cloud_defend/privilege_escalation_debugfs_launched_inside_a_privileged_container.toml:30]:
```suggestion
severity = "medium"
```

**Review by DefSecSentinel — APPROVED** [review]
LGTM

**Pull request merged**